packages/@aws-cdk/aws-stepfunctions/README.md

```diff
@@ -458,6 +458,22 @@ const activity = new stepfunctions.Activity(this, 'Activity');
 new cdk.CfnOutput(this, 'ActivityArn', { value: activity.activityArn });
 ```
 
+### Activity-Level Permissions
+
+Granting IAM permissions to an activity can be achieved by calling the `grant(principal, actions)` API:
+
+```ts
+const activity = new stepfunctions.Activity(this, 'Activity');
+
+const role = new iam.Role(stack, 'Role', {
+  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+});
+
+activity.grant(role, 'states:SendTaskSuccess');
+```
```

Comment:
Should we have a `grantAll()` methods? I feel like we are going to save a lot of copy pasta for users. @shivlaks what do you think?

Reply:
i do see the value in reducing copy-pasta but would say it’s better to revisit the need at a later time. i’d be a little reluctant to add the `grantAll` mainly because:

* `StateMachine` and `Activity` are 2 different resources in the Step Functions namespace - activities can be referenced inside of a state machine but both an exist independently, similar to Lambda functions, SQS queues, SNS topics, etc. These resources all offer their own grant APIs and that’s how permissions are managed today.
* `all` is bound to carry a different meaning at a different point in time aside from the context in which it’s running. For example, when a new feature is added, we would need to update all and that new feature might not be pertinent to users.
  * I’d counter myself here by saying it might be okay since using `all` might mean the intent is to opt into everything without having to make future code changes. However, we haven’t done this anywhere else in the construct library yet either.

Reply:
I get the not wanting to use the `grantAll()` terminology, but I still think we are going to see this code a lot:

```typescript
activity.grant(role, 'states:SendTaskSuccess')
activity.grant(role, 'states:DescribeActivity')
activity.grant(role, 'states:DeleteActivity')
activity.grant(role, 'states:GetActivityTask')
activity.grant(role, 'SendTaskFailure')
activity.grant(role, 'states:SendTaskHeartbeat')
```
Which can be reduced by having a more specific grant method, that being said I do agree we can hold this off until we get some customers feedback.

Reply:
gotcha, I'd expect it to be a single call to grant with the list of permitted actions as a single parameter i.e. `activity.grant(role, ['states:sentTaskSuccess', 'states:DescribeActivity', ...])`

or logical groupings of APIs. You're still right that there will likely be a couple of grant calls that we can consolidate down the road.